### DISCUSSION TITLE
v5.3.5

### DISCUSSION THREAD

**Discussion by auvipy** [celery/celery#8631] [General]
Main theme of this release is adding Python 3.12 compatibility support to all through the projects dependencies. Also lots of bugs were squashed. Dependencies upgraded and docs improved.

## What's Changed
* Update test.txt versions by @auvipy in https://github.com/celery/celery/pull/8481
* fix os.getcwd() FileNotFoundError by @mortimer2015 in https://github.com/celery/celery/pull/8448
* Fix typo in CONTRIBUTING.rst by @monteiro-renato in https://github.com/celery/celery/pull/8494
* typo(doc): configuration.rst by @shifenhutu in https://github.com/celery/celery/pull/8484
* assert before raise by @monteiro-renato in https://github.com/celery/celery/pull/8495
* Update GHA checkout version by @auvipy in https://github.com/celery/celery/pull/8496
* Fixed replaced_task_nesting by @Nusnus in https://github.com/celery/celery/pull/8500
* Fix code indentation for route_task() example by @stefmolin in https://github.com/celery/celery/pull/8502
* support redis 5.x by @dulmandakh in https://github.com/celery/celery/pull/8504
* Fix typos in test_canvas.py by @monteiro-renato in https://github.com/celery/celery/pull/8498
* Marked flaky tests by @Nusnus in https://github.com/celery/celery/pull/8508
* Fix typos in calling.rst by @visitorckw in https://github.com/celery/celery/pull/8506
* Added support for replaced_task_nesting in chains by @Nusnus in https://github.com/celery/celery/pull/8501
* Fix typos in canvas.rst by @visitorckw in https://github.com/celery/celery/pull/8509
* Patch Version Release Checklist by @Nusnus in https://github.com/celery/celery/pull/8488
* Added Python 3.11 support to Dockerfile by @Nusnus in https://github.com/celery/celery/pull/8511
* Dependabot (Celery) by @Nusnus in https://github.com/celery/celery/pull/8510
* Bump actions/checkout from 3 to 4 by @dependabot in https://github.com/celery/celery/pull/8512
* [pre-commit.ci] pre-commit autoupdate by @pre-commit-ci in https://github.com/celery/celery/pull/8515
* Update ETA example to include timezone by @amantri in https://github.com/celery/celery/pull/8516
* Replaces datetime.fromisoformat with the more lenient dateutil parser by @stumpylog in https://github.com/celery/celery/pull/8507
* Fixed indentation in Dockerfile for Python 3.11 by @Nusnus in https://github.com/celery/celery/pull/8527
* Fix git bug in Dockerfile by @Nusnus in https://github.com/celery/celery/pull/8528
* Tox lint upgrade from Python 3.9 to Python 3.11 by @Nusnus in https://github.com/celery/celery/pull/8526
* Document gevent concurrency by @cunla in https://github.com/celery/celery/pull/8520
* Update test.txt by @auvipy in https://github.com/celery/celery/pull/8530
* Celery Docker Upgrades by @Nusnus in https://github.com/celery/celery/pull/8531
* pyupgrade upgrade v3.11.0 -> v3.13.0 by @Nusnus in https://github.com/celery/celery/pull/8535
* Update msgpack.txt by @auvipy in https://github.com/celery/celery/pull/8548
* Update auth.txt by @auvipy in https://github.com/celery/celery/pull/8547
* Update msgpack.txt to fix build issues by @auvipy in https://github.com/celery/celery/pull/8552
* Basic ElasticSearch / ElasticClient 8.x Support by @q2justin in https://github.com/celery/celery/pull/8519
* Fix eager tasks does not populate name field by @KOliver94 in https://github.com/celery/celery/pull/8486
* [pre-commit.ci] pre-commit autoupdate by @pre-commit-ci in https://github.com/celery/celery/pull/8559
* Fix typo in celery.app.control by @Spaceface16518 in https://github.com/celery/celery/pull/8563
* Update solar.txt ephem by @auvipy in https://github.com/celery/celery/pull/8566
* Update test.txt pytest-timeout by @auvipy in https://github.com/celery/celery/pull/8565
* Correct some mypy errors by @rbtcollins in https://github.com/celery/celery/pull/8570
* [pre-commit.ci] pre-commit autoupdate by @pre-commit-ci in https://github.com/celery/celery/pull/8572
* Update elasticsearch.txt by @auvipy in https://github.com/celery/celery/pull/8573
* Update test.txt deps by @auvipy in https://github.com/celery/celery/pull/8574
* [pre-commit.ci] pre-commit autoupdate by @pre-commit-ci in https://github.com/celery/celery/pull/8587
* Update test.txt by @auvipy in https://github.com/celery/celery/pull/8590
* Improved the "Next steps" documentation (#8561). by @frolenkov-nikita in https://github.com/celery/celery/pull/8600
* Disabled couchbase tests due to broken package breaking main by @Nusnus in https://github.com/celery/celery/pull/8602
* Update elasticsearch deps by @auvipy in https://github.com/celery/celery/pull/8605
* Update cryptography==41.0.5 by @auvipy in https://github.com/celery/celery/pull/8604
* Update pytest==7.4.3 by @auvipy in https://github.com/celery/celery/pull/8606
* test initial support of python 3.12.x by @auvipy in https://github.com/celery/celery/pull/8549
* updated new versions to fix CI by @auvipy in https://github.com/celery/celery/pull/8607
* Update zstd.txt by @auvipy in https://github.com/celery/celery/pull/8609
* Fixed CI Support with Python 3.12 by @Nusnus in https://github.com/celery/celery/pull/8611
* updated CI, docs and classifier for next release by @auvipy in https://github.com/celery/celery/pull/8613
* updated dockerfile to add python 3.12 by @auvipy in https://github.com/celery/celery/pull/8614
* lint,mypy,docker-unit-tests -> Python 3.12 by @Nusnus in https://github.com/celery/celery/pull/8617
* Correct type of `request` in `task_revoked` documentation by @RJPercival in https://github.com/celery/celery/pull/8616
* update docs docker image by @auvipy in https://github.com/celery/celery/pull/8618
* Fixed RecursionError caused by giving `config_from_object` nested mod… by @frolenkov-nikita in https://github.com/celery/celery/pull/8619
* Fix: serialization error when gossip working by @kitsuyui in https://github.com/celery/celery/pull/6566
* [documentation] broker_connection_max_retries of 0 does not mean "retry forever" by @jakila in https://github.com/celery/celery/pull/8626
* added 2  debian package for better stability in Docker by @auvipy in https://github.com/celery/celery/pull/8629
* Added changelog for v5.3.5 by @auvipy in https://github.com/celery/celery/pull/8623

## New Contributors
* @mortimer2015 made their first contribution in https://github.com/celery/celery/pull/8448
* @monteiro-renato made their first contribution in https://github.com/celery/celery/pull/8494
* @shifenhutu made their first contribution in https://github.com/celery/celery/pull/8484
* @stefmolin made their first contribution in https://github.com/celery/celery/pull/8502
* @visitorckw made their first contribution in https://github.com/celery/celery/pull/8506
* @dependabot made their first contribution in https://github.com/celery/celery/pull/8512
* @amantri made their first contribution in https://github.com/celery/celery/pull/8516
* @cunla made their first contribution in https://github.com/celery/celery/pull/8520
* @q2justin made their first contribution in https://github.com/celery/celery/pull/8519
* @Spaceface16518 made their first contribution in https://github.com/celery/celery/pull/8563
* @rbtcollins made their first contribution in https://github.com/celery/celery/pull/8570
* @frolenkov-nikita made their first contribution in https://github.com/celery/celery/pull/8600
* @RJPercival made their first contribution in https://github.com/celery/celery/pull/8616
* @kitsuyui made their first contribution in https://github.com/celery/celery/pull/6566
* @jakila made their first contribution in https://github.com/celery/celery/pull/8626

**Full Changelog**: https://github.com/celery/celery/compare/v5.3.4...v5.3.5

<hr /><em>This discussion was created from the release <a href='https://github.com/celery/celery/releases/tag/v5.3.5'>v5.3.5</a>.</em>

**Comment by auvipy** [comment]
If any breaking change or regression found please report ASAP.

**Reply by auvipy** [reply]
Back ports of bugs or hot fixes should be applied in https://github.com/celery/celery/tree/v5.3 branch if not possible in main branch. no new features only big fixes critical for production.

**Comment by Nusnus** [comment]
@auvipy Just reporting that the ongoing smoke tests using the pytest-celery have passed on v5.3.5 running on Python 3.12.
This is mostly an update that this upgrade passes smoothly with everything pytest-celery/smoke-tests related.

```
============================================================ 41 passed, 2 xfailed in 136.66s (0:02:16) ============================================================
.pkg: _exit> python /Users/nusnus/dev/GitHub/celery/.tox/.tox/lib/python3.12/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
  3.12-smoke: OK (138.43=setup[1.00]+cmd[137.42] seconds)
  congratulations :) (138.66 seconds)
```

**Reply by auvipy** [reply]
Oh yeah! Can you guide me how to use that as well?

**Reply by auvipy** [reply]
let us wait till 15th november to start merging 5.4 features

**Reply by Nusnus** [reply]
@auvipy 
> Oh yeah! Can you guide me how to use that as well?

When it's ready you'll be able to learn from the official documentation :)

> let us wait till 15th november to start merging 5.4 features

Cool!

**Reply by Nusnus** [reply]
> When it's ready you'll be able to learn from the official documentation :)

We can also have a 1:1 session if you wish for a quick course on the new project when it reaches the proper maturity stage.

**Reply by auvipy** [reply]
that would be really great

**Comment by matejkloska** [comment]
@auvipy it seems like after upgrading from v5.3.4 to v5.3.5 we had a growing number of `Messages in flight` for SQS broker and no messages were acknowledged. Once we downgraded back, all appeared again fine. Is known issue or a new bug?

**Reply by auvipy** [reply]
It is a botocore issue. We need to pin that to 1.31.80 or less

**Reply by matejkloska** [reply]
Please, can you provide more details? We have `boto3==1.28.30` and `botocore==1.31.78` and it's not working too.

**Reply by auvipy** [reply]
I got this from here https://github.com/celery/celery/discussions/8627

**Reply by matejkloska** [reply]
Alright, I'll wait for a resolution there. Thanks!
